### DISCUSSION TITLE
SIGNIN_OAUTH_ERROR on google provider

### DISCUSSION THREAD

**Discussion by tcetin** [nextauthjs/next-auth#6090] [Help]
### Question 💬

Hi,

I am getting this error when I try to log in with google social provider.

`TypeError [ERR_INVALID_ARG_TYPE]: The "listener" argument must be of type function. Received an instance of Object.`

It works fine if I don't visit any other page except the home page (http://localhost:3000). But if I visit any other page after signing in except the home page, then  I sign out on any page after that; when I try to sign in, it throws this error. 

Here are my google login provider settings:

```
GoogleProvider({
  clientId: process.env.NEXT_PUBLIC_GOOGLE_CLIENT_ID,
  clientSecret: process.env.NEXT_PUBLIC_GOOGLE_CLIENT_SECRET,
  authorization: {
    params: {
      prompt: 'consent',
      access_type: 'offline',
      response_type: 'code',
    },
  },
}),

```
I also use NEXTAUTH_SECRET in the next auth configuration:

`secret: process.env.NEXTAUTH_SECRET`

When I debug the application, it throws this error in openid-client package that is in use for getting the issuer:

<img width="996" alt="image" src="https://user-images.githubusercontent.com/5071753/208300880-aa73f3dc-be8e-4012-ade5-c0791e06e6bc.png">

### How to reproduce ☕️

1. Login with google provider 
2. Visit another page and then sign out
3. Try to log in again by google provider

### Contributing 🙌🏽

Yes, I am willing to help answer this question in a PR

**Comment by tcetin** [comment]
Is there any idea about the problem?

**Reply by ivanhueso** [reply]
Hi, I'm getting the same issue. And i cannot figure out why is this happening.

**Reply by kloppster** [reply]
Also getting this issue. Anyone find a solution or reason it is happening?

**Reply by imprakharshukla** [reply]
Did you manage to solve it by any chance?

**Reply by mikeacre** [reply]
I am also receiving this issue but only on certain devices, and randomly. It appears on certain devices, after a successful login, after about a day or so they will no longer be able to login. I am running this as production on an AWS server. ~10-15 users for internal use app using google oAuth for login only. 

I know it must have something to do with the tokens not refreshing for certain users. Is there a way to force this refresh? Or maybe I  missed something?

Full Error:

```
[next-auth][error][SIGNIN_OAUTH_ERROR] 
 https://next-auth.js.org/errors#signin_oauth_error The "listener" argument must be of type function. Received an instance of Object {
   error: {
     message: 'The "listener" argument must be of type function. Received an instance of Object',
     stack: 'TypeError [ERR_INVALID_ARG_TYPE]: The "listener" argument must be of type function. Received an instance of Object\n' +
       '    at checkListener (node:events:266:3)\n' +
       '    at ClientRequest.once (node:events:647:3)\n' +
       '    at new ClientRequest (node:_http_client:237:10)\n' +
       '    at Object.request (node:https:357:10)\n' +
       '    at Object.<anonymous> (/home/ubuntu/git_dirs/lmsign-full/next-display/node_modules/proxy-agent/node_modules/agent-base/patch-core.js:25:22)\n' +
       '    at /home/ubuntu/git_dirs/lmsign-full/next-display/node_modules/socks-proxy-agent/node_modules/agent-base/patch-core.js:23:20\n' +
       '    at Function.request (/home/ubuntu/git_dirs/lmsign-full/next-display/node_modules/openid-client/lib/helpers/request.js:120:73)\n' +
       '    at Function.discover (/home/ubuntu/git_dirs/lmsign-full/next-display/node_modules/openid-client/lib/issuer.js:144:38)\n' +
       '    at openidClient (/home/ubuntu/git_dirs/lmsign-full/next-display/node_modules/next-auth/core/lib/oauth/client.js:16:41)\n' +
       '    at getAuthorizationUrl (/home/ubuntu/git_dirs/lmsign-full/next-display/node_modules/next-auth/core/lib/oauth/authorization-url.js:70:49)',
     name: 'TypeError'
   },
   providerId: 'google',
   message: 'The "listener" argument must be of type function. Received an instance of Object'
 }

```

Next Auth:
```
`GoogleProvider({
            clientId: process.env.GOOGLE.CLIENT_ID,
            clientSecret: process.env.GOOGLE.SECRET,
            callbackUrl: "HIDDEN",
            authorization: {
                params: {
                    prompt: "consent",
                    access_type: "offline",
                    response_type: "code"
                }
            }

        }),
`
```

**Comment by timeisenhuth** [comment]
This started recently happening for me as well randomly and now its basically unusable

**Comment by MaciejJezierskiFiskars** [comment]
I had this issue recently. I am glad this topic wasn't closed as another 2. I had problem when I included mailgun-js library. When I removed this from the imports it basically started to work again. It has occured on Google, Facebook and Discord. I am assuming some library is used in different versions which is causing an issue when trying to establish connection with 3rd party. Hope it will navigate to root of the problem.

My stack was 
Google
`The "listener" argument must be of type function. Received an instance of Object\n    at __node_internal_captureLargerStackTrace (node:internal/errors:497:5)\n    at new NodeError (node:internal/errors:406:5)\n    at __node_internal_ (node:internal/validators:440:11)\n    at checkListener (node:events:274:3)\n    at ClientRequest.once (node:events:648:3)\n    at new ClientRequest (node:_http_client:239:10)\n    at Object.request (node:https:378:10)\n    at Object.<anonymous> (/Users/maciej/Projects/me/wk-app/node_modules/agent-base/patch-core.js:25:22)\n    at /Users/maciej/Projects/me/wk-app/node_modules/socks-proxy-agent/node_modules/agent-base/patch-core.js:23:20\n    at Issuer.request (/Users/maciej/Projects/me/wk-app/node_modules/openid-client/lib/helpers/request.js:123:73)\n    at Issuer.discover (/Users/maciej/Projects/me/wk-app/node_modules/openid-client/lib/issuer.js:143:36)\n    at openidClient (/Users/maciej/Projects/me/wk-app/node_modules/next-auth/core/lib/oauth/client.js:16:41)\n    at getAuthorizationUrl (/Users/maciej/Projects/me/wk-app/node_modules/next-auth/core/lib/oauth/authorization-url.js:70:49)\n    at Object.signin (/Users/maciej/Projects/me/wk-app/node_modules/next-auth/core/routes/signin.js:38:60)\n    at AuthHandler (/Users/maciej/Projects/me/wk-app/node_modules/next-auth/core/index.js:260:39)\n    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at Object.handler (/Users/maciej/Projects/me/wk-app/node_modules/@sidebase/nuxt-auth/dist/runtime/server/services/authjs/nuxtAuthHandler.mjs:88:24)\n    at async file:///Users/maciej/Projects/me/wk-app/node_modules/h3/dist/index.mjs:1962:19\n    at async Object.callAsync (file:///Users/maciej/Projects/me/wk-app/node_modules/unctx/dist/index.mjs:72:16)\n    at async Server.toNodeHandle (file:///Users/maciej/Projects/me/wk-app/node_modules/h3/dist/index.mjs:2249:7)',
     name: 'TypeError' },
  providerId: 'google',
  message:
   'The "listener" argument must be of type function. Received an instance of Object' }
`

Facebook
`[next-auth][error][OAUTH_CALLBACK_ERROR] 
https://next-auth.js.org/errors#oauth_callback_error The "listener" argument must be of type function. Received an instance of Object { error:
   { TypeError [ERR_INVALID_ARG_TYPE]: The "listener" argument must be of type function. Received an instance of Object
       at __node_internal_captureLargerStackTrace (node:internal/errors:497:5)
       at new NodeError (node:internal/errors:406:5)
       at __node_internal_ (node:internal/validators:440:11)
       at checkListener (node:events:274:3)
       at ClientRequest.once (node:events:648:3)
       at new ClientRequest (node:_http_client:239:10)
       at Object.request (node:https:378:10)
       at Object.<anonymous> (/Users/maciej/Projects/me/wk-app/node_modules/agent-base/patch-core.js:25:22)
       at /Users/maciej/Projects/me/wk-app/node_modules/socks-proxy-agent/node_modules/agent-base/patch-core.js:23:20
       at Client.request (/Users/maciej/Projects/me/wk-app/node_modules/openid-client/lib/helpers/request.js:123:73)
       at Client.authenticatedPost (/Users/maciej/Projects/me/wk-app/node_modules/openid-client/lib/helpers/client.js:187:18)
       at async Client.grant (/Users/maciej/Projects/me/wk-app/node_modules/openid-client/lib/client.js:1343:22)
       at async Client.oauthCallback (/Users/maciej/Projects/me/wk-app/node_modules/openid-client/lib/client.js:620:24)
       at async oAuthCallback (/Users/maciej/Projects/me/wk-app/node_modules/next-auth/core/lib/oauth/callback.js:111:16)
       at async Object.callback (/Users/maciej/Projects/me/wk-app/node_modules/next-auth/core/routes/callback.js:52:11)
       at async AuthHandler (/Users/maciej/Projects/me/wk-app/node_modules/next-auth/core/index.js:208:28)
       at Object.handler (/Users/maciej/Projects/me/wk-app/node_modules/@sidebase/nuxt-auth/dist/runtime/server/services/authjs/nuxtAuthHandler.mjs:88:24)
       at async file:///Users/maciej/Projects/me/wk-app/node_modules/h3/dist/index.mjs:1962:19
       at async Object.callAsync (file:///Users/maciej/Projects/me/wk-app/node_modules/unctx/dist/index.mjs:72:16)
       at async Server.toNodeHandle (file:///Users/maciej/Projects/me/wk-app/node_modules/h3/dist/index.mjs:2249:7)
     name: 'OAuthCallbackError',
     code: 'ERR_INVALID_ARG_TYPE' },
  providerId: 'facebook',
  message:
   'The "listener" argument must be of type function. Received an instance of Object' }
`

**Reply by russellr922** [reply]
It was mailgun-js for us as well, thank you for sharing.

**Comment by dansaim** [comment]
I found the culprit.
One of my module's dependencies was overriding node's default http/s request with its own implementation, which included an old version of https-proxy-agent which had this code that was causing the problem so I forced it to use the latest one by putting this in my project's package.json:
`  "resolutions": {
    "https-proxy-agent": "^7.0.5"
  },`

**Reply by dansaim** [reply]
Mind you my issue was with Keycloak as a provider, not Facebook or Google but it seems to be the same issue - 2 vs 3 arguments.
